soccer/gameplay/tactics/positions/submissive_defender.py

```diff
@@ -29,10 +34,45 @@ def __init__(self):
 
         self.add_state(SubmissiveDefender.State.marking,
                        behavior.Behavior.State.running)
+        self.add_state(SubmissiveDefender.State.clearing,
+                       behavior.Behavior.State.running)
 
         self.add_transition(behavior.Behavior.State.start,
                             SubmissiveDefender.State.marking, lambda: True,
                             "immediately")
+        self.add_transition(SubmissiveDefender.State.marking,
+                            SubmissiveDefender.State.clearing,
+                            lambda: self.go_clear,
+                            "when it is safe to clear the ball")
+        self.add_transition(
+            SubmissiveDefender.State.clearing,
+            SubmissiveDefender.State.marking,
+            lambda: self.subbehavior_with_name('kick-clear').state == behavior.Behavior.State.completed or not self.should_clear_ball(evaluation.ball.time_to_ball(self.robot)),
+            "done clearing")
+
+    def should_clear_ball(self, our_time_to_ball):
+        #Returns true if our robot can reach the ball sooner than the closest opponent
+        if main.ball().pos.mag() < constants.Field.ArcRadius * 2:
```

Comment:
How about something like this:
* remove any logic related to determining if the ball should be cleared from SubmissiveDefender and put it all in Defense.  You could have a `should_clear_ball()` in Defense that returns True if the ball is within radius and one of our two defending robots is closer (time-wise) than the closest opponent.  The SubmissiveDefender behavior will then unconditionally go for the ball when `go_clear` is set and all responsibility for deciding when the robot should clear will be solely up to the Defense behavior.
* Remove the Defense.go_clear variable and change the transitions to and from `clearing` to call the `should_clear_ball()` method.
* Remove most of `execute_clearing` - that logic should now be in `should_clear_ball()`
* Make the on_enter_clearing and on_exit_clearing methods set the go_clear variable for the defenders

I think this will simplify things a bit and will consolidate the logic for deciding when to clear the ball.  Thoughts?